morphir/src/org/finos/morphir/datamodel/Defaults.scala

```diff
@@ -0,0 +1,121 @@
+package org.finos.morphir.datamodel
+import zio.*
+
+type DefaultTask = Task[Data]
+
+sealed trait MDMDefaultError                        extends Throwable
+case class Unimplemented(msg: String)               extends MDMDefaultError
+case class NoDefaultNothing()                       extends MDMDefaultError
+case class Unfillable(data: Data, concept: Concept) extends MDMDefaultError
+
+trait Defaults {
```

Comment:
I'd do this as a typeclass instead of an interface

Reply:
Considering these are just defaults to existing Concepts, I'd do them in their corresponding case object instead of an extra interface.

Reply:
My only problem with that is that they're recursively defined - users may want to use the `defaultRecord`'s behavior for **how** a record should be filled with default values, but need to provide their own definition for what the default of a particular type is. Resolving that - so that `morphir-scala`'s `defaultRecord` code will make recursive calls to a user-defined `defaultLocalDate` function - is the reason I structured it this way.